sdv/cag/fixed_increments.py

```diff
@@ -0,0 +1,218 @@
+            The increment that each value in the column must be a multiple of. Must be greater
+            than 0 and a whole number.
+        table_name (str, optional):
+            The name of the table that contains the column. Optional if the
+            data is only a single table. Defaults to None.
+    """
+
+    @staticmethod
+    def _validate_init_inputs(column_name, increment_value, table_name):
+        if not isinstance(column_name, str):
+            raise ValueError('`column_name` must be a string.')
+        if increment_value <= 0:
+            raise ValueError('`increment_value` must be greater than 0.')
+        if increment_value % 1 != 0:
+            raise ValueError('`increment_value` must be a whole number.')
+
+        _validate_table_name_if_defined(table_name)
+
+    def __init__(self, column_name, increment_value, table_name=None):
+        super().__init__()
+        self._validate_init_inputs(column_name, increment_value, table_name)
+        self.column_name = column_name
+        self.table_name = table_name
+        self._fixed_increments_column_name = f'{self.column_name}#increment'
+        self.increment_value = increment_value
+        self._dtype = None
+
+    def _validate_pattern_with_metadata(self, metadata):
+        """Validate the pattern is compatible with the provided Metadata.
+
+        Validates that:
+            - If no table_name is set, checks that the Metadata only contains a single table
+            - The column_name exist in the table in the Metadata
+            - The column_name is a numerical sdtype
+        Args:
+            metadata (sdv.metadata.Metadata):
+                The input Metadata to validate.
+        """
+        _validate_table_and_column_names(
+            self.table_name, columns=[self.column_name], metadata=metadata
+        )
+        table_name = self._get_single_table_name(metadata)
+        col_sdtype = metadata.tables[table_name].columns[self.column_name]['sdtype']
+        if col_sdtype != 'numerical':
+            raise PatternNotMetError(
+                f"Column '{self.column_name}' has an incompatible sdtype ('{col_sdtype}')."
+                "The column sdtype must be 'numerical'."
+            )
+
+    def _validate_pattern_with_data(self, data, metadata):
+        """Validate the data is compatible with the pattern.
+
+        Args:
+            data (dict[pd.DataFrame]):
+                The data to validate
+
+            metadata (sdv.metadata.Metadata):
+                The input Metadata to use to validate the data.
+
+        Returns:
+            None
+        """
+        valid = self._check_if_divisible(
+            data, self._get_single_table_name(metadata), self.column_name, self.increment_value
+        )
+        if not valid.all():
+            invalid_rows_str = _get_invalid_rows(valid)
+            raise PatternNotMetError(
+                'The fixed increments requirement has been met because the data is not '
```

Comment:
typo: has *not* been met

Reply:
Fixed:
https://github.com/sdv-dev/SDV/pull/2409/commits/ab7486461cdacaa4a32ab55016ba3204f780c406#diff-442749eb6608ea424c56131f8b5c2633ab037f4f758317361cc150224d2d4620R96